cmd/migrate/main_test.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"testing"
+
+	fakedownstreammachineclientset "github.com/kubermatic/machine-controller/pkg/client/clientset/versioned/fake"
+	machinev1alpha1downstream "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
+
+	fakeclusterv1alpha1clientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/fake"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestMigrateMachines(t *testing.T) {
+	testcases := []struct {
+		DownStreamMachine *machinev1alpha1downstream.Machine
+		Node              *corev1.Node
+		ErrExpected       bool
+	}{
+		{
+			DownStreamMachine: &machinev1alpha1downstream.Machine{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-machine"},
+				Spec:       machinev1alpha1downstream.MachineSpec{ObjectMeta: metav1.ObjectMeta{Name: "test-node"}},
+			},
+			Node: &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "test-node"}},
+		},
+		// We currently fail if no corresponding node exists. Is this desired?
```

Comment:
Given that the machine resource is healthy no errors and deletion hasn't been scheduled this is valid assumption.

Reply:
Well yeah but this is supposed to work even when a machine was just created or is misconfigured. 

I think I should change this to not require a corresponding node to exist because since it runs in an `initContainer` it will block any further reconciliation, meaning if you happen to deploy this while a machine did not have a corresponding node manual intervention is required. WDYT?

Reply:
yeah, I think you are right.